src/components/NewChatForm.tsx

```diff
@@ -2,28 +2,51 @@ import { useMessagingStore } from "../store/messaging.store";
 import { useWalletStore } from "../store/wallet.store";
 import React, { useState, useCallback, useEffect, useRef } from "react";
 import { kaspaToSompi, sompiToKaspaString } from "kaspa-wasm";
+import { KaspaAddress } from "./KaspaAddress";
 import styles from "../components/NewChatForm.module.css";
 
 interface NewChatFormProps {
   onClose: () => void;
+  prefilledRecipient?: string;
+  prefilledResolvedAddress?: string;
+  prefilledDomainId?: string;
 }
 
-export const NewChatForm: React.FC<NewChatFormProps> = ({ onClose }) => {
-  const [recipientAddress, setRecipientAddress] = useState("");
+export const NewChatForm: React.FC<NewChatFormProps> = ({
+  onClose,
+  prefilledRecipient = "",
+  prefilledResolvedAddress = "",
+  prefilledDomainId = "",
+}) => {
+  const [recipientAddress, setRecipientAddress] = useState(prefilledRecipient);
   const [handshakeAmount, setHandshakeAmount] = useState("0.2");
   const [error, setError] = useState<string | null>(null);
   const [recipientWarning, setRecipientWarning] = useState<string | null>(null);
   const [isLoading, setIsLoading] = useState(false);
-  const [showConfirmation, setShowConfirmation] = useState(false);
+  const [showConfirmation, setShowConfirmation] = useState(
+    !!(prefilledRecipient && prefilledResolvedAddress)
+  );
   const [isCheckingRecipient, setIsCheckingRecipient] = useState(false);
 
   // kns related
-  const [resolvedAddress, setResolvedAddress] = useState<string | null>(null);
+  const [resolvedAddress, setResolvedAddress] = useState<string | null>(
+    prefilledResolvedAddress || null
+  );
   const [isResolvingKns, setIsResolvingKns] = useState(false);
   const [knsError, setKnsError] = useState<string | null>(null);
-  const [knsDomainId, setKnsDomainId] = useState<string | null>(null);
+  const [knsDomainId, setKnsDomainId] = useState<string | null>(
+    prefilledDomainId || null
+  );
   const knsDomainRef = useRef<string>("");
 
+  // Helper to determine if input is a Kaspa address
+  const isKaspaAddress =
+    recipientAddress.startsWith("kaspa:") ||
+    recipientAddress.startsWith("kaspatest:");
```

Comment:
this can be memoized:
```ts
const isRecipientValueAnAddressFormat = useMemo(() => {
  return (
    // expression here
  )
}, [recipientAddress])